lms/djangoapps/ccx/api/v0/views.py

```diff
@@ -156,9 +158,47 @@ def get_valid_input(request_data, ignore_missing=False):
             field_errors['max_students_allowed'] = {'error_code': 'invalid_max_students_allowed'}
     elif 'max_students_allowed' in request_data:
         field_errors['max_students_allowed'] = {'error_code': 'null_field_max_students_allowed'}
+
+    course_modules = request_data.get('course_modules')
+    if course_modules is not None:
+        if isinstance(course_modules, list):
+            for course_module_id in course_modules:
+                try:
+                    UsageKey.from_string(course_module_id)
+                except InvalidKeyError:
+                    field_errors['course_modules'] = {'error_code': 'invalid_course_module_keys'}
+                    break
+            else:
+                valid_input['course_modules'] = course_modules
+        else:
+            field_errors['course_modules'] = {'error_code': 'invalid_course_module_list'}
+    elif 'course_modules' in request_data:
+        # case if the user actually passed null as input
+        valid_input['course_modules'] = None
+
     return valid_input, field_errors
 
 
+def valid_course_modules(course_module_list, master_course_key):
+    """
+    Function to validate that each element in the course_module_list belongs
+    to the master course structure.
+    Args:
+        course_module_list (list): A list of strings representing Block Usage Keys
+        master_course_key (CourseKey): An object representing the master course key id
+
+    Returns:
+        bool: whether or not all the course module strings belong to the master course
+    """
+    try:
+        course_structure_obj = CourseStructure.objects.get(course_id=master_course_key)
+    except CourseStructure.DoesNotExist:
+        return False
+    course_structure = course_structure_obj.structure
+    course_chapters = course_structure['blocks'][course_structure['root']].get('children', [])
```

Comment:
Since this logic is used in the test code as well *and* because logic defines the paid unit of a CCX course, it'd be nice to move it out into a well-defined method that both places use.